cmd/nlb_create.go

```diff
@@ -43,22 +40,26 @@ func (c *nlbCreateCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
 }
 
 func (c *nlbCreateCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	nlb := &egoscale.NetworkLoadBalancer{
-		Description: utils.NonEmptyStringPtr(c.Description),
-		Labels: func() (v *map[string]string) {
-			if len(c.Labels) > 0 {
-				return &c.Labels
-			}
-			return
-		}(),
-		Name: &c.Name,
+	nlb := v3.CreateLoadBalancerRequest{
+		Description: c.Description,
+		Labels:      c.Labels,
+		Name:        c.Name,
 	}
+	var err error
 
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, v3.ZoneName(c.Zone))
```

Comment:
Feel free to change some type in all struct definition if you want to match egoscale v3 one, like for `ZoneName`, since everything will be v3

Reply:
it could be applied to all the other type you see and think relevant to be updated

Reply:
updated in 4abefb6f